internal/driver/glfw/canvas.go

```diff
@@ -162,29 +164,45 @@ func (c *glCanvas) Refresh(obj fyne.CanvasObject) {
 }
 
 func (c *glCanvas) Focus(obj fyne.Focusable) {
-	if c.focused != nil {
-		c.focused.FocusLost()
+	c.RLock()
```

Comment:
Could this be simplified to a single lock/unlock?

```go
c.Lock()
prev := c.focused
c.focused = obj
c.Unlock()

if prev != nil {
    prev.FocusLost()
}
if obj != nil {
    obj.FocusGained()
}
```
Ditto for Unfocus

Reply:
It could, but somehow it felt "right" to notify FocusLost before the new focus item was set... Maybe I was overly cautious?

Reply:
It is currently two lines longer than the suggestion, and besides that I don't think there is anything to be gained so let's keep it as is